File: sesac7_flutter_recipe_app/lib/data/mapper/recipes_mapper.dart
import 'package:flutter_recipe_app/data/dto/recipes_dto.dart';
import 'package:flutter_recipe_app/data/mapper/recipe_mapper.dart';
import 'package:flutter_recipe_app/data/model/recipes.dart';

extension RecipesDtoToModel on RecipesDto {
  Recipes toModel() {
    return Recipes(
      recipes: recipes?.map((recipe) => recipe.toModel()).toList() ?? [],
    );
  }
}

extension RecipesModelToDto on Recipes {
  RecipesDto toDto() {
    return RecipesDto(
      recipes: recipes.map((model) => model.toDto()).toList(),
    );
  }
}

File: compose_recipe/app/src/main/java/com/example/recipeapp/data/mapper/RecipesMapper.kt
package com.example.recipeapp.data.mapper

import com.example.recipeapp.data.dto.RecipesDto
import com.example.recipeapp.data.model.Recipes

fun RecipesDto.toModel(): Recipes {
    return Recipes(
        recipes = this.recipes?.map { it.toModel() } ?: emptyList()
    )
}

fun Recipes.toDto(): RecipesDto {
    return RecipesDto(
        recipes = this.recipes.map { it.toDto() }
    )
}